### ISSUE TITLE
Shadow problems since version 1.7.1

### ISSUE THREAD

**Issue by FaicSilier** [x3dom/x3dom#616]
Hello,
When I update from 1.7 to 1.7.1 the shadows are not working correctly anymore.
But they work again if a zoom out enough.
Screenshots attached.

1.7.1 shadows don't work if i'm too near
![b](https://cloud.githubusercontent.com/assets/16774386/12739078/0916e5c6-c9a1-11e5-808e-2fd6eb777309.PNG)
1.7.1 they start working if i zoom out enough
![b-zoomedout](https://cloud.githubusercontent.com/assets/16774386/12739079/093f68fc-c9a1-11e5-8adb-fdcf9137443d.PNG)
1.7 same scene just with 1.7, everything works
![a](https://cloud.githubusercontent.com/assets/16774386/12739080/0942cdda-c9a1-11e5-8a89-ae0b5496e849.PNG)

What could be the problem and how can I fix it?

**Comment by FaicSilier** [comment]
I made a example scene. You can see the flickering of the edges on the roof. This was not happening before 1.7.1

Example scene:
https://gist.github.com/FaicSilier/0e2f659dc22e32998117

**Comment by andreasplesch** [comment]
Live:
https://rawgit.com/FaicSilier/0e2f659dc22e32998117/raw/274163cceea4813a7e770f00b99cc39f14f00fa5/asd.html

**Comment by andreasplesch** [comment]
On FF 38.5 on linux I do not see obvious flickering in the example scene ?

![vrmath2x4d](https://cloud.githubusercontent.com/assets/6171115/13056646/91a62a6c-d3e4-11e5-87f3-e7becca38111.png)

Shadows look ok to me ?

**Comment by FaicSilier** [comment]
Do the shadows also look ok from different distances? For me they are ok when I'm very close or very far away but in between they flicker around. (See the right edge of the roof)
As mentioned this didn't happens on version 1.7 and smaller. So 1.7.1 introduced the bug.

![capture](https://cloud.githubusercontent.com/assets/16774386/13065543/65ca0094-d498-11e5-9041-8db04a77486d.PNG)

**Comment by andreasplesch** [comment]
Yes, I can confirm seeing these edge effects. It looks like adding an Appearance would not help ?
Any suspicion which change after 1.7 caused this ?

**Comment by FaicSilier** [comment]
I don't know how to look through the changes from 1.7 to 1.7.1 ... I played around with every parameter but none had an effect on this problem except using no lights so no shadow.

I have to emphasize that this is a major problem and should be fixed as soon as possible.

**Comment by mlimper** [comment]
@tsturm Do you think this could be connected with the dynamic shadow shader?
https://github.com/x3dom/x3dom/commit/5e6386d09283574c78c1547fc7645b3fdc3c813f

@all If anyone finds the time to check, using the version before and after this commit, that could be very helpful

**Comment by andreasplesch** [comment]
Using 1.7.2 release the example looks like this on linux chrome and ff:
![image](https://cloud.githubusercontent.com/assets/6171115/25484859/9f009b1e-2b29-11e7-8ad2-1655f1d251a2.png)
I cannot see flickering or edge effects on the roof. Can this issue be closed ?
